x-pack/plugins/beats/server/lib/adapters/database/kibana_database_adapter.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FrameworkRequest } from '../famework/adapter_types';
+import {
+  DatabaseAdapter,
+  DatabaseBulkIndexDocumentsParams,
+  DatabaseCreateDocumentParams,
+  DatabaseCreateDocumentResponse,
+  DatabaseDeleteDocumentParams,
+  DatabaseDeleteDocumentResponse,
+  DatabaseGetDocumentResponse,
+  DatabaseGetParams,
+  DatabaseIndexDocumentParams,
+  DatabaseKbnESCluster,
+  DatabaseKbnESPlugin,
+  DatabaseMGetParams,
+  DatabaseMGetResponse,
+  DatabasePutTemplateParams,
+  DatabaseSearchParams,
+  DatabaseSearchResponse,
+} from './adapter_types';
+
+export class KibanaDatabaseAdapter implements DatabaseAdapter {
+  private es: DatabaseKbnESCluster;
+
+  constructor(kbnElasticSearch: DatabaseKbnESPlugin) {
+    this.es = kbnElasticSearch.getCluster('admin');
+  }
+  public async putTemplate(
+    req: FrameworkRequest | null,
+    params: DatabasePutTemplateParams
+  ): Promise<any> {
+    const callES = this.getCallType(req);
```

Comment:
@weltenwort I considered that. The request object is very far from easy to return null, as it is the very essence of hapi. If that were to be hacked all bets are off anyway.

That said I think an argument can be made perhaps for readability to use version 1 of your suggestion?

Reply:
It's not just `null`, but any value that is coerced to `false`. Hapi does not need to be hacked, a simple bug is enough.

I would just recommend to be a bit more defensive here since it effectively is "the permission check" for all Elasticsearch requests performed via this adapter. 

Yes, variant 1 proposed above should already protect from that.

Reply:
OK, I will make that change. Even if just for readability I think its worth it

Reply:
fixed